### DISCUSSION TITLE
`dagger config` polish

### DISCUSSION THREAD

**Discussion by sipsma** [dagger/dagger#12440] [Ideas]
(Follow ups from https://github.com/dagger/dagger/pull/6533 related to `dagger config`)

`dagger config` currently only prints the configuration of a module, needs to be expanded with subcommands for actually setting the values not settable elsewhere. *Currently* those values are just:
* `include`
* `exclude`

But we should keep in mind that list may grow over time.

Quoting @shykes from the previous PR around some possibilities here:
- `git config` style. I don't remember what it is, but it exists so worth using as a baseline :)
- Specialized map keys to subcommands:
     - `dagger config`: print configuration
     - `dagger config set KEY VALUE`: set a configuration key. Note, making `KEY` a subcommand makes it enumerable.
     - `dagger config get KEY`: print the value of a key
- Not allow generic config crud, and use specialized commands instead, for example `dagger develop --sdk` to change the SDK field; `dagger install` to change the Dependencies field, etc. Then just have `dagger info` to print information about the current Dagger configuration - including the contents of your config, but also engine version, etc.

---

Tentatively adding to `v0.10` milestone since it would very nice to at least be able configure include/exclude via the CLI by then. The workaround in the mean time is to hand edit `dagger.json`.

**Comment by levlaz** [comment]
Based on #7324 I think we should also add some more docs to this page around what the possible key values are. https://docs.dagger.io/reference/979596/cli/#dagger-config
